core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.java

```diff
@@ -40,7 +41,7 @@
  */
 
 @Singleton
-public class SharedPreferenceUtil {
+public class SharedPreferenceUtil implements Serializable {
```

Comment:
Do not make anything serializable in android ever. If you need to pass stuff around you use parcelable and even so you should not be passing this class or DialogShower around.

Serializable has awful performance impacts on Android.